### PR TITLE
Switch to jest

### PR DESCRIPTION
**Pull request by kingsleyzissou**
- Switched all the tests from AVA to Jest
- Tests can be run from root level or package level
- Had to add a build stage for circleci

### CODE REVIEW

**Comment by wtrocki** [comment]
@kingsleyzissou Can you point me to the problem you have mentioned. I forgot where to look. On local machine all tests are passing. Proposed structure works fine. It looks like this is also going to resolve couple other package oriented issues. We can now rely on the coverage. Great timing for that changes as there is no major PR created that will affect it

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,6 +22,9 @@ jobs:
       - run:
           name: run lint
           command: yarn lint
+      - run:
```
Comment (wtrocki) [inline .circleci/config.yml]: I was assuming that test will do prebuild of the packages. Good catch

File: .vscode/launch.json [hunk 2]
```diff
@@ -152,5 +152,22 @@
       ],
       "cwd": "${workspaceRoot}/templates/ts-apollo-fullstack",
     },
+    {
```
Comment (wtrocki) [inline .vscode/launch.json]: ❤️

File: packages/graphback-cli/tests/workflow-test.ts [hunk 3]
```diff
@@ -25,30 +24,52 @@ const model = {
...
-  t.true(existsSync(join(basePath, "client/src/graphql/queries/findNotes.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/createNote.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/createComment.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/deleteComment.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/deleteNote.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/updateComment.graphql")));
-  t.true(existsSync(join(basePath, "client/src/graphql/mutations/updateNote.graphql")));
-  t.true(existsSync(join(basePath, "src/schema/schema.graphql")));
-  t.true(existsSync(join(basePath, "src/resolvers/resolvers.ts")));
+  expect(existsSync(join(basePath, "db.sqlite"))).toBe(true);
```
Comment (wtrocki) [inline packages/graphback-cli/tests/workflow-test.ts]: Oh.. This must have been so hard. 
We can improve this by snapshotting folder structure.

File: packages/graphback-core/tests/__snapshots__/EngineTests.ts.snap [hunk 4]
```diff
@@ -0,0 +1,21 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (wtrocki) [inline packages/graphback-core/tests/__snapshots__/EngineTests.ts.snap]: Sadly Snapshots in Jest are much less readable :(. Nothing to do here anyway

File: .circleci/config.yml (hunk 1 above)
Comment (kingsleyzissou) [reply .circleci/config.yml]: I removed the pretest script, it didn't seem to be working

**Comment by kingsleyzissou** [comment]
@wtrocki it's the graphback-cli tests. It doesn't cause the tests to fail, but the tests don't run. Here's a screenshot from circleci:

image

File: packages/graphback-core/tests/__snapshots__/EngineTests.ts.snap (hunk 4 above)
Comment (kingsleyzissou) [reply packages/graphback-core/tests/__snapshots__/EngineTests.ts.snap]: I can try see if I can enable the markdown too.

File: packages/graphback-cli/tests/workflow-test.ts (hunk 3 above)
Comment (kingsleyzissou) [reply packages/graphback-cli/tests/workflow-test.ts]: This is the test that didn't really run, so can't confirm if it actually works. Also, that part was done by the jest-codemods.

It was some of the other tests that I had to do manually

**Comment by wtrocki** [comment]
Oh yes. This was happening before so not something new, due to db migration logic. I will fix that in separate PR.

**Pull request merged**